app/controllers/application_controller.rb

```diff
@@ -1215,6 +1215,11 @@ def get_view(db, options = {}, fetch_data = false)
       params[:sortby]      = view.headers.index(params[:sort_choice])
     end
 
+    report_symbols = [:all_sortcol, :savedreports_tree_sortcol, :reports_tree_sortcol]
+    if session[sortcol_sym].try(:to_i).nil? && report_symbols.include?(sortcol_sym)
```

Comment:
`session[sortcol_sym].try(:to_i).nil?`

what exactly is this checking?  it seems like it's only checking for nil (or is it also checking for abnormal objects?)

Reply:
remember that `nil.to_i` returns `0`, so probably want to check `session[sortcol_sym].nil?`

Reply:
Its just checking for a nil value, if nil then the default column sort is used

Reply:
also, any string whose contents are non-numeric will convert to 0 as well.  So, `"42".to_i` is `42`, while `"wednesday".to_i` is `0`

Reply:
re-reading what is coded, code is doing a to_i conversion and making sure that is not nil and then checking that the symbol (prior to the to_i conversion) is included in an array.  Does not seem to make sense

Reply:
Yes this makes more sense.

Reply:
I think what you have coded reflects your words, but a programmer later in this code may be trying to figure out this logic.  Maybe a comment to explain the rationale for this seemingly odd condition would be helpful.